elastic/kibana
# [Console Monaco Migration] Fix variable replacement

Fixes https://github.com/elastic/kibana/issues/185999

## Summary

This PR fixes the variable replacement when the variable contains an object value .

**How to test:**
1. Go to Console and create the following variables:
`var1`: `{"match_all": {}}`
`var2`: `index`
`var3`: `match_all`
2. Test using an object variable inside the request data:
```
GET _search
{
  "query": "${var1}"
}
```
should be sent as `query: {match_all: {}}` in the payload.
3. Test using a variable in the URL:
`PUT /test-${var2}` should be sent as `PUT /test-index` in the payload.
4. Test using a string variable in the request data:
```
GET _search
{
  "query": {
    "${var3}": {}
  }
}
```
should be sent as `query: {match_all: {}}` in the payload.
5. Test using undefined variables:
```
GET _search
{
  "query": {
    "${var4}": {}
  }
}
```
or `PUT /test-${var5}`
Both should be sent as they are.
